src/app/services/user/user.service.ts

```diff
@@ -29,11 +27,31 @@ export class UserService {
 
     let serviceBase = this.delApiServices;
 
+    if (null == userId) {
```

Comment:
Please fix this segment - you don't need to rebuild the entire thing. Use the conditional to check if you need to append the userId. Else return the url